sacc/tracer.py

```diff
@@ -17,7 +17,17 @@ def __init__ (self, name, type, z, Nz, exp_sample=None, Nz_sigma_logmean=None,
         self.sigma_logmean=Nz_sigma_logmean
         self.sigma_logwidth=Nz_sigma_logwidth
         self.DNz=DNz
+        self.Mproxy_name=str(Mproxy_name)
+        self.Mproxy_min=Mproxy_min
+        self.Mproxy_max=Mproxy_max
         self.extra_cols={}
+
+        if (not(self.Mproxy_name==str(None))) and (self.type != "spin0"):
```

Comment:
The right way to compare against None in python is `if (self.Mproxy_name is not None) and (self.type != "spin0")`, etc.

Reply:
The most correct thing to do is to NOT do this line a few above: `self.Mproxy_name=str(Mproxy_name)`, which sets the Mproxy_name to be the string "None". You can leave it as a `None` and then put in what Anze suggested.

Reply:
Yes, this caused a bug actually... I fixed it.